### PR TITLE
Add ComputeLobprob Tests

### PR DESCRIPTION
**Pull request by Jack-Khuu**
Moves `compute_logprobs` from GRPO app to `utils/ops.py` and adds basic tests for it

Also updated the logit slicing math, to an equivalent version based on TRL

Old
```
    context_length = logits.shape[1] - input_ids.shape[1]
    logits = logits[:, context_length - 1 : -1].to(input_ids.device)
```
New
```
    logits = logits[:, -input_ids.size(1) - 1 : -1, :].to(input_ids.device)
```

---
` python -m apps.grpo.main --config apps/grpo/qwen3_1_7b.yaml`

**wandb** (main): torchforge/grpo-training/runs/gq9wjime
image

**wandb** (PR): torchforge/grpo-training/runs/ctirsyoa
image

` pytest tests/unit_tests/util/test_compute_logprobs.py`

### CODE REVIEW

File: src/forge/util/ops.py [hunk 1]
```diff
@@ -49,3 +49,16 @@ def selective_log_softmax(logits: torch.Tensor, index: torch.Tensor) -> torch.Te
             per_token_logps.append(row_per_token_logps)
         per_token_logps = torch.stack(per_token_logps)
     return per_token_logps
+
+
+def compute_logprobs(
+    logits: torch.Tensor, input_ids: torch.Tensor, temperature: float = 1.0
+) -> torch.Tensor:
```
Comment (joecummings) [inline src/forge/util/ops.py:56]: Can you document the params in the docstring as well?

File: src/forge/util/ops.py [hunk 2]
```diff
@@ -49,3 +49,16 @@ def selective_log_softmax(logits: torch.Tensor, index: torch.Tensor) -> torch.Te
...
+    """
+    Computes the log probabilities of the input tokens given the model logits and temperature.
+    """
+    context_length = logits.shape[1] - input_ids.shape[1]
```
Comment (joecummings) [inline src/forge/util/ops.py]: nit since were here: I think you can do `logits = logits[:, -input_ids.size(1): -1]` and not have to compute the context length above

Comment (Jack-Khuu) [reply src/forge/util/ops.py]: Uhhhh I'm gonna need to see the math on this, this doesn't smell right

Comment (joecummings) [reply src/forge/util/ops.py]: https://github.com/huggingface/trl/blob/417915a3e4d3e3bc8d7b196594308b8eabf928be/trl/trainer/grpo_trainer.py#L829

L829 -> L831

Comment (Jack-Khuu) [reply src/forge/util/ops.py]: Ok the math is looking less sus, but the loss doesn't seem to be happy with it OOTB so I think we make this a different PR instead of squeezing it in

**torchforge/grpo-training/runs/w5i55g7q** (current)
image

**torchforge/grpo-training/runs/b6sh1jnj**(TRL style)
image

**Comment by casteryh** [comment]
Can you revive some of these tests? https://github.com/meta-pytorch/forge/blob/0e1984195c39816082c8d770f8186dfb6f4be764/tests/unit_tests/actors/test_reference_actor.py#L17-L98

File: src/forge/util/ops.py [hunk 3]
```diff
@@ -49,3 +49,25 @@ def selective_log_softmax(logits: torch.Tensor, index: torch.Tensor) -> torch.Te
             per_token_logps.append(row_per_token_logps)
         per_token_logps = torch.stack(per_token_logps)
     return per_token_logps
+
+
+def compute_logprobs(
+    logits: torch.Tensor, input_ids: torch.Tensor, temperature: float = 1.0
+) -> torch.Tensor:
```
Comment (casteryh) [inline src/forge/util/ops.py:56]: can you add a warning saying that the prompt lengths have to be the same in the batch for this to work?

**Comment by Jack-Khuu** [comment]
> Can you revive some of these tests?

Did not realize these tests still existed. yup will revive

File: src/forge/util/ops.py (hunk 2 above)
Comment (Jack-Khuu) [reply src/forge/util/ops.py]: Nvn, math is indeed mathing converting to TRL style (little tweak)
